skimage/exposure/exposure.py

```diff
@@ -341,6 +341,9 @@ def rescale_intensity(image, in_range='image', out_range='dtype'):
     array([  0,  63, 127], dtype=int8)
 
     """
+    if image.min() == image.max():
```

Comment:
I feel like you should use `imin` and `imax` computed below. The check should likely be done just before line 354 is computed. Where by you compare the denominator to `0` to check if you are dividing by zero.

Reply:
That makes sense. Just to clarify, we should return the clipped image instead of the original image right?

Reply:
Yes i believe so. The output would therefore be `omin` after all the math is performed.

I generally think we shouldn't spend too much time special casing edge cases like this, but not returning `nan` makes sense.

Reply:
Sounds good, I've updated my PR anyways. 

There is a failed test from Travis but is seems unrelated to the fix.

Reply:
These two lines should be removed right?